src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -473,6 +456,62 @@ public Long getAllowedTokenExpirationClockSkewSeconds() {
         return allowedTokenExpirationClockSkewSeconds;
     }
 
+    @PostConstruct
+    @Restricted(NoExternalUse.class)
+    public void createProxyAwareResourceRetriver() {
+        proxyAwareResourceRetriever =
+                ProxyAwareResourceRetriever.createProxyAwareResourceRetriver(isDisableSslVerification());
+    }
+
+    ProxyAwareResourceRetriever getResourceRetriever() {
+        return proxyAwareResourceRetriever;
+    }
+
+    private OidcConfiguration buildOidcConfiguration() {
+        // TODO cache this and use the well known if available.
+        OidcConfiguration conf = new OidcConfiguration();
+        conf.setClientId(clientId);
+        conf.setSecret(clientSecret.getPlainText());
+
+        // TODO what do we prefer?
+        // conf.setPreferredJwsAlgorithm(JWSAlgorithm.HS256);
+        // set many more as needed...
+
+        OIDCProviderMetadata oidcProviderMetadata = serverConfiguration.toProviderMetadata();
+        if (this.isDisableTokenVerification()) {
+            conf.setAllowUnsignedIdTokens(true);
+            conf.setTokenValidator(new AnythingGoesTokenValidator());
+        }
+        conf.setProviderMetadata(oidcProviderMetadata);
+        if (oidcProviderMetadata.getScopes() != null) {
+            // auto configuration does not need to supply scopes
+            conf.setScope(oidcProviderMetadata.getScopes().toString());
+        }
+        conf.setUseNonce(!this.nonceDisabled);
+        if (allowedTokenExpirationClockSkewSeconds != null) {
+            conf.setMaxClockSkew(allowedTokenExpirationClockSkewSeconds.intValue());
+        }
+        conf.setResourceRetriever(getResourceRetriever());
+        if (this.isPkceEnabled()) {
+            conf.setPkceMethod(CodeChallengeMethod.S256);
+        }
+        return conf;
+    }
+
+    @Restricted(NoExternalUse.class) // exposed for testing only
+    protected OidcClient buildOidcClient() {
+        OidcConfiguration oidcConfiguration = buildOidcConfiguration();
+        OidcClient client = new OidcClient(oidcConfiguration);
+        // add the extra settings for the client...
+        client.setCallbackUrl(buildOAuthRedirectUrl());
+        client.setAuthenticator(new OidcAuthenticator(oidcConfiguration, client));
```

Comment:
yikes..  `OidcAuthenticator` ignroes the clients configuration for `resourceRetreiver` so ignores the Jenkins proxy and settings to disable TLS validation!

Reply:
347dd4da345a52783110abfbe7a0a880dfb935d0